### DISCUSSION TITLE
Entries with a taxonomy that span a collection

### DISCUSSION THREAD

**Discussion by wesort** [statamic/cms#6091] [Troubleshooting]
On an entry's permalink, I'm having trouble returning entries across multiple collections that share a taxonomy. These are effectively "related entries". This may be a templating / syntax error on my part, or a feature request.

Context:
- Collections: `Films`, `Stills`, `Other`
- Taxonomy: 'Sub-Collections' which has this term: `London` (and others)
- `Sub-Collections` portion of entry's content:
```
sub_collections:
  - london
  - miami
``` 

When on the taxonomy route `/sub-collections/london` the expected entries are returned: _all_ entries across collections that have that `sub_collections` term `london`.
Template: `/views/sub_collections/show.antlers.html`:
```
{{ entries }}
    <div>
        {{ results }}
            <a href="{{ url }}">
                {{ title }}
            </a>
        {{ /results }}
    </div>
{{ /entries }}
```

When on an entry's permalink, _only_ the entries with that term from within the _same collection_, _not_ all entries across the collections with that term. I've tried templating this in a few ways:
Templating attempted: `views/archive/show.antlers.html`
```
{{ sub_collections }}
    <h2{{ title }}</h2>
    <div>
        {{ entries }}
            {{ if id != page:id }}
                <a href="{{ url }}">
                    {{ title }}
                </a>
            {{ /if }}
        {{ /entries }}
    </div>
{{ /sub_collections }}
```

I've also tried this but it returns an exception of "Modifier [from] not found":
```
{{ collection from="films|stills|other" :taxonomy:sub_collections="sub_collections" }}
    <h2>{{ title }}</h2>
    <div>
        {{ entries }}
            {{ if id != page:id }}
                <a href="{{ url }}">
                    {{ title }}
                </a>
            {{ /if }}
        {{ /entries }}
    </div>
{{ /collection }}
```

## From [Neue Taxonomies #74](https://github.com/statamic/cms/issues/74#issue-384867048)
> **5. When in the context of a Collection, we only care about its own Terms.**
In other words, when we're designing the tags and parameters, you will only see the terms that exist inside that Collection unless you explicitly say otherwise.

Any ideas?

**Comment by edalzell** [comment]
What version and what parser are you using?

**Reply by wesort** [reply]
Statamic: `v3.3.12`
`config/statamic/antlers.php`: `'version' => 'runtime',`

**Reply by JohnathonKoster** [reply]
For the `modifier [from] not found`, there must be an injected `collection` variable taking priority. You can force the tag like so (https://statamic.dev/new-antlers-parser#disambiguating-tags):

```antlers
{{ %collection from="film|stills|other" :taxonomy:sub_collections="sub_collections" }}
    <h2>{{ title }}</h2>
    <div>
        {{ entries }}
            {{ if id != page:id }}
                <a href="{{ url }}>
                    {{ title }}
                </a>
            {{ /if }}
        {{ /entries }}
    </div>
{{ /%collection }}
```

**Reply by wesort** [reply]
Thanks @JohnathonKoster.

It's taken me a lot of trial and error (and it smells messy), but I think I've gotten there now:
- I needed to wrap the `%collection` tag in the `sub_collections` tag and wrap that in an `if` statement
- I also needed to set `:taxonomy:sub_collections="slug"`

```
{{ if sub_collections }}
    {{ sub_collections }}
        <section">
            <h2 id="sub-collection-{{ slug }}">
                <a href="#sub-collection-{{ slug }}">{{ title }}</a>
            </h2>
            {{ %collection from="films|stills|other" :taxonomy:sub_collections="slug" as="entries" }}
                <div>
                    {{ entries }}
                        {{ if id != page:id }}
                            {{ title }}
                        {{ /if }}
                    {{ /entries }}
                </div>
            {{ /%collection }}
        </section>
    {{ /sub_collections }}
{{ /if }}
```

Any suggestions for a cleaner approach would be welcome.
